### PR TITLE
PLAT-576: fix race in SMTestAPICall. Add tests

### PR DESCRIPTION
**Pull request by ivanshibitov**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: ledger-core/application/testwalletapi/statemachine/statemachine_test.go [hunk 1]
```diff
@@ -49,42 +53,114 @@ func TestSMTestAPICall_Migrate(t *testing.T) {
...
+		// check that we remove bargein after migrate
+		_, bargeIn = slotMachine.SlotMachine.GetPublishedGlobalAliasAndBargeIn(outgoingRef)
+		require.Nil(t, bargeIn)
+	}
+
+	slotMachine.RunTil(smWrapper.AfterStep(smRequest.stepRegisterBargeIn))
+	newOutgoingCall := smRequest.requestPayload.CallOutgoing
+	var newBargeIn smachine.BargeInHolder
+	{
+		newOutgoingRef := reference.NewRecordOf(APICaller, newOutgoingCall)
```
Comment (bigbes) [inline ledger-core/application/testwalletapi/statemachine/statemachine_test.go:90]: can you add one check: `outgoingRef` != `newOutgoingRef`?

Comment (ivanshibitov) [reply ledger-core/application/testwalletapi/statemachine/statemachine_test.go:90]: There is, little bit lower 
require.NotEqual(t, outgoingCall, newOutgoingCall)

Comment (ivanshibitov) [reply ledger-core/application/testwalletapi/statemachine/statemachine_test.go:90]: no, it's direct check

Comment (ivanshibitov) [reply ledger-core/application/testwalletapi/statemachine/statemachine_test.go:90]: done

File: ledger-core/application/testwalletapi/statemachine/statemachine.go [hunk 2]
```diff
@@ -94,7 +89,19 @@ func (s *SMTestAPICall) stepRegisterBargeIn(ctx smachine.ExecutionContext) smach
 	if !ctx.PublishGlobalAliasAndBargeIn(outgoingRef, bargeInCallback) {
 		return ctx.Error(errors.New("failed to publish bargeInCallback"))
 	}
-	return ctx.Jump(s.stepSendRequest)
+	s.registeredBargeInRef = outgoingRef
```
Comment (cyraxred) [inline ledger-core/application/testwalletapi/statemachine/statemachine.go:92]: to pass a one-time param you can use a closure and do not pollute SM's body, e.g.

outgoingRef := ...
return ctx.Jump(func (ctx) ... {
   // use outgoingRef
})

**Pull request merged**